r/R/dplyr-funcs-datetime.R

```diff
@@ -147,5 +147,15 @@ register_bindings_datetime <- function() {
   register_binding("pm", function(x) {
     !call_binding("am", x)
   })
-
+  register_binding("tz", function(x) {
+    if (call_binding("is.Date", x)) {
+      abort("timezone extraction for objects of class `date` not supported in Arrow")
+    } else if (call_binding("is.numeric", x)) {
+      abort("timezone extraction for objects of class `numeric` not supported in Arrow")
+    } else if (call_binding("is.character", x)) {
+      abort("timezone extraction for objects of class `character` not supported in Arrow")
+    } else {
+      return(x$type()$timezone())
+    }
```

Comment:
TIL `...$ToString()`. I really like your suggestion. I thought of something like it and couldn't figure out how exactly to implement it. I guess the only limitation is that we will now surface the arrow data types instead of the matching R ones, which might cause confusion.

Reply:
Done.